### PR TITLE
StyleCI integration

### PR DESCRIPTION
**Pull request by Soullivaneuh**
I propose you to integrate [StyleCI](https://styleci.io/) on your project.

With this tool, you will have automatic code style scanning on PR and push to prevent code style errors.

Things you have to do:

* Enable this bundle on StyleCI
* Set your header comment on web interface configuration
* ~~Ping me. Then I'll apply CS and rebase the PR.~~ Already done. :+1: 

I suggest Symfony preset. If you want to change it, don't hesitate to ping me. :+1:

### CODE REVIEW

File: .styleci.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+preset: symfony
+
+enabled:
+  - psr0
```
Comment (theofidry) [inline .styleci.yml]: we shouldn't enable PSR-0 should we?

File: composer.json [hunk 2]
```diff
@@ -22,7 +22,8 @@
     "require-dev": {
         "doctrine/common": "~2.3",
         "symfony/property-access": "~2.2",
-        "phpunit/phpunit": "3.7.*"
```
Comment (theofidry) [inline composer.json]: It might be worth to remove PHPUnit form the dev package?

Comment (Soullivaneuh) [inline composer.json]: I didn't remove it.

But It should be removed IMO. But this is the place for another PR! ;-)

File: .styleci.yml (hunk 1 above)
Comment (Soullivaneuh) [inline .styleci.yml]: For this, I just copy selected fixers enabled before on `.php_cs` file.

**Comment by Seldaek** [comment]
I really don't like failing builds due to CS checks. Maintainers can run php-cs-fixer once in a while to fix the little details that get through every now and then, but bothering people with patch files when they try to contribute something isn't achieving anything IMO.

**Comment by Soullivaneuh** [comment]
> Maintainers can run php-cs-fixer once in a while to fix the little details that get through every now and then

I not agree with you. Running regularly CS fixes will pollute git history IMHO.

And BTW, I think it's also a good think to learn CS conventions to contributors.

Any chance to get your opinion changed about this PR?

Thanks.

**Comment by Soullivaneuh** [comment]
And BTW, see the changes. Lot of files corrected with CS.

This is the proof that running php-cs-fixer "regularly" is not the good way for me. :-1:

**Comment by theofidry** [comment]
What's good with StyleCI is that it's a dedicated service so it separates well your travis build with continuous integration from a code style build.

**Comment by Soullivaneuh** [comment]
@Seldaek If I can add one more argument, StyleCI has a nice system to propose automatic patches.

If a PR is wrong with CS, contributors just has to click on fix button and accept the sub PR if they don't want to run a single command line. :+1:

**Comment by Seldaek** [comment]
Lots of minimal changes yes, not all of them I would include in the fixer config.. one CS fix commit every now and then pollutes less than CS fix commits in every PR I'd argue, but in any case my point isn't about git history purity or codebase purity, it's about making it a smoother experience for people that contribute, because not all of them are familiar with the gazillion build tools and blinking red lights we add to all repos. This can be pretty overwhelming for newcomers and does not help the open source cause.

**Comment by theofidry** [comment]
As long as it's run once in a while like before each release it's good for me ;)

I agree that two much new tools may overwhelm newcomers. I'm less favorable for this when dealing with bundles as it targets already more advanced developers, but for non specific framework libraries like this it may ease the developer experience.

**Comment by Soullivaneuh** [comment]
> less than CS fix commits in every PR

Command line and squash are here for that. :-)

> because not all of them are familiar with the gazillion build tools and blinking red lights

As I said: 

> And BTW, I think it's also a good think to learn CS conventions to contributors.

If every repository has this politics, those contributors would never learn about this. This is how I learned for my personal case.

I understand your point of view, but I still think this is a big miss to avoid this kind of tools.

**Comment by theofidry** [comment]
> This is how I learned for my personal case.

Same for me actually.

Another thing possible is to put it but not register StyleCI as *required* for PR. That don't prevent the failed status of a PR, but still indicates it's a non mandatory element.

**Comment by Soullivaneuh** [comment]
ping @Seldaek

**Comment by Seldaek** [comment]
I said what I had to say.. we can't agree on everything, it's life, but thankfully it's about a trivial matter of coding style, not actual functionality. If @tshelburne thinks otherwise and wants to merge ok. I gave him carte blanche on this repo and I'll happily be overruled, but if it's left up to me it won't go in.

**Comment by Soullivaneuh** [comment]
> I said what I had to say.. we can't agree on everything, it's life

Sure. I just wanted to know your opinion about our arguments. ;-)

> If @tshelburne thinks otherwise and wants to merge ok. I gave him carte blanche on this repo

In this case, maybe you can reopen this PR.

**Comment by Seldaek** [comment]
Well he'll get the messages and reopen if needed I'd say.

**Comment by tshelburne** [comment]
I agree with @Seldaek on this one - it's already a challenge to keep OS repos going, and having as few impediments as possible is pretty crucial. It's certainly not that CS isn't important, but I consider it much less important than having a living codebase and contributor pool - and we *do* have tools to fix problems "regularly" as they show up.

**Comment by Soullivaneuh** [comment]
If this is your politics, I delete this branch then.

Thanks for your attention.

**Pull request closed without merging**